src/reducer/cart-wishlist-reducer.js

```diff
@@ -0,0 +1,99 @@
+// export function cartReducer() {
+//    
+export function cartAndWishlistReducer(state, action) {
+    console.log("inside dispatch");
+    switch (action.type) {
+        case "SET_CART":
+            return {
+                ...state,
+                cartList: [...action.payload]
+            }
+        case "SET_WISHLIST":
+            return {
+                ...state,
+                wishList: [...action.payload]
+            }
+
+        case "ADD_TO_CART":
+            console.log(action.payload);
+            return {
+                ...state,
+                cartList: [...state.cartList, { ...action.payload, quantity: 1 }]
+            };
+        case "SEARCH_ITEM_IN_CART":
+            console.log(action.payload);
+            return {
+                ...state,
+                cartList: state.cartList.filter(
+                    (cartListItem) => cartListItem.name === action.payload
+                )
+            };
+        case "ADD_TO_WISHLIST":
+            console.log("inside add to wishlist", action.payload);
+            return {
+                ...state,
+                wishList: state.wishList.concat(action.payload)
+            };
+        case "REMOVE_FROM_WISHLIST":
+            console.log(action.payload);
+
+            return {
+                ...state,
+                wishList: state.wishList.filter(
+                    (wishListItem) => wishListItem._id !== action.payload._id
+                )
+            };
+
+        case "ADD_TO_CART_FROM_WISHLIST":
+            console.log(action.payload);
+            return {
+                ...state,
+                cartList: [...state.cartList, { ...action.payload, quantity: 1 }],
+                wishList: state.wishList.filter(
+                    (wishListItem) => wishListItem._id !== action.payload._id
+                )
```

Comment:
This should not be needed .
ADD_TO_CART should work same... from Products lisiting page or Wishlist page